Fix model cache and its resetting

kostobog opened the pull request:
@blcham 
Fixes #415

kostobog commented on s-pipes-core/src/main/java/cz/cvut/spipes/manager/OntoDocManager.java:
```diff
@@ -138,6 +141,34 @@ public OntModel getOntology(String uri) {
+     * Cache is implemented by OntDocumentManager.getInstance().getFileManager(). Additionally, jena may store models in
+     * OntModelSpec.OWL_MEM.getBaseModelMaker and OntModelSpec.OWL_MEM.getImportModelMaker. Specifically, the
+     * ImportModelMaker stores imports and it is necessary to clear
+     * (TODO - it is necessary if FileManager is not set for caching. Verify that this holds when FileManager is set for caching.
```
TODO comment removed. The comment was not written correctly and is no longer relevant.
`FileManager` is now always set for caching, i.e. regardless of value of `reloadFiles` parameter.

kostobog commented on s-pipes-core/src/main/java/cz/cvut/spipes/manager/OntoDocManager.java:
```diff
@@ -89,6 +90,8 @@ private OntoDocManager() {
     public static OntologyDocumentManager getInstance() {
         if (sInstance == null) {
             sInstance = new OntoDocManager(OntDocumentManager.getInstance());
+            // TODO - not tested!!!
```
Comment removed.

blcham commented on s-pipes-core/src/test/java/cz/cvut/spipes/manager/OntoDocManagerTest.java:
```diff
@@ -4,33 +4,219 @@
+        o1ModelAfterChange = ontoDocManager.getOntology(o1.getURI());
+        o2ImportedAfterChange = o1ModelAfterChange.getImportedModel(o2.getURI());
+
+        assertFalse(o1ModelBeforeChange.hashCode() == o1ModelAfterChange.hashCode());
+        assertFalse(o1ModelBeforeChange.isIsomorphicWith(o1ModelAfterChange));
```
```suggestion
        assertSameModel(o1ModelBeforeChange, o1ModelAfterChange);
```

blcham replied:
you are repeating those 2 lines multiple times ...  how about implementing it in private method that would be named appropriately

kostobog replied:
Refactored assert patterns into reusable methods.

blcham commented on s-pipes-core/src/test/java/cz/cvut/spipes/manager/OntoDocManagerTest.java:
```diff
@@ -4,33 +4,219 @@
+        assertTrue(o2ImportedBeforeChange.isIsomorphicWith(o2ImportedAfterChange));
+
+
+        // check cache updated if reloadFiles is true
```
I believe this is in conflict with test method name `ontModelsNotReloadedWhenKeepUpdatedIsFalse` .. shouldn't we split it into mutliptle methods ?

blcham commented on s-pipes-core/src/test/java/cz/cvut/spipes/manager/OntoDocManagerTest.java:
```diff
@@ -4,33 +4,219 @@
+        OntModel o1ModelAfterChange = ontoDocManager.getOntology(o1.getURI());
+        OntModel o2ImportedAfterChange = o1ModelAfterChange.getImportedModel(o2.getURI());
+
+        // CHECK - o1 model retrieved before and after change is the same when cache not updated. Check is done by o1
+        // model objects hash and by the contents of the models
```
there are many comments in this test ... can we clear and unify them ... 

what big CHECK means compared to `check cache updated if reloadFiles is true` ... is this CHECK suppose to be TODO ?

blcham reviewed (changes requested):
I am afraid this added functionality will hit us back sooner or later since we use deprecated API ...  thus it is very good that you tested it !

However, tests are very hard to read and contains too many strange comments .. Fastest is probably to ask chatbot to improve it. Some test methods are checking multiple things so mostlikely we should split them.

blcham commented on s-pipes-core/src/test/java/cz/cvut/spipes/manager/OntoDocManagerTest.java:
```diff
@@ -1,36 +1,233 @@
+     * @throws Exception
+     */
+    @Test
+    public void testOntoDocManager_registerDocuments_CacheAndItsUpdateBasedOnReloadFilesValue() throws Exception {
```
```suggestion
    public void registerDocumentsWithReloadFilesFlagCorreclyUpdatesSourceAndCachedOntologies() throws Exception {
```

blcham replied:
It seems like you are not obeying conventions ... : 
- 1) remove prefix testOntoDocManager_ ... as test methods should not start with "test" prefix and from class OntoDocManagerTest.java it is clear that we are testing OntoDocManager.java  ...
- 2) consider my suggestion if it makes sense

blcham commented on s-pipes-core/src/test/java/cz/cvut/spipes/manager/OntoDocManagerTest.java:
```diff
@@ -1,36 +1,233 @@
+     * @throws Exception
+     */
+    @Test
+    public void testOntoDocManager_registerDocuments_CacheAndItsUpdateBasedOnReloadFilesValue_updatingIntermediateImportedModelsImports() throws Exception {
```
```suggestion
    public void registerDocuments_CacheAndItsUpdateBasedOnReloadFilesValue_updatingIntermediateImportedModelsImports() throws Exception {
```

blcham replied:
see comment above

kostobog commented on s-pipes-core/src/main/java/cz/cvut/spipes/manager/OntoDocManager.java:
```diff
@@ -213,11 +285,24 @@ public static Map<String, Model> getAllFile2Model(Path directoryOrFilePath) {
+    private static void addSHACLRelevantModel(String fileName, Model model) {
         String baseURI = JenaUtils.getBaseUri(model);
 
         if (baseURI != null) {
```
Removed

kostobog commented on s-pipes-core/src/main/java/cz/cvut/spipes/manager/OntoDocManager.java:
```diff
@@ -61,18 +61,21 @@ public class OntoDocManager implements OntologyDocumentManager {
     private static boolean reloadFiles = false;
 
-    // TODO remove !!!!!!! this is workaround for registering SPIN related things.
-    private static Model allLoadedFilesModel = ModelFactory.createDefaultModel();
+    private static Map<String, Model> allChangedFiles = new HashMap<>();
+    private static Set<Path> removedFiles = new HashSet<>();
+    private static Set<Path> managedFiles;
```
Removed

kostobog commented:
@blcham 
[Commit](https://github.com/kbss-cvut/s-pipes/pull/417/commits/f09250bc0d0bfe363e0a90401d7b44c17e1f30d1) fixes handing of unchanged renamed files, see also [comment ](https://github.com/kbss-cvut/s-pipes/pull/417#discussion_r2354867424) above. I added ([commit](https://github.com/kbss-cvut/s-pipes/pull/417/commits/0e3d26e1e301f705e988b484154ec152f2b7592a)) a test to verify such file system changes were properly handled. 

Additionally, one of the tests was failing because of `OntDocumentManager` IRI to path mapping was polluted from by other tests. This issue was resolved in [commit](https://github.com/kbss-cvut/s-pipes/pull/417/commits/e3d9a5ff22623af1cf08ddea8b4f928116081115) by resetting the mapping in the `OntDocumentManager.reset` method. 

I also did some minor changes in javadoc. There some unused methods which I think can be deleted but for now added TODO to delete because I am not sure if they are not used somewhere else, e.g. spipes editor.
